tests/cbtest/gtest/test_export.cc

```diff
@@ -0,0 +1,267 @@
+       return (0);
+}
+
+/*
+ * if you find the need to copy paste the following
+ * two functions consider to put them in a generic
+ * header file, because I already copied them =)
+ *
+ */
+
+std::string getCmdPath(std::string zfsCmd) {
+	std::string cmdPath;
+	const char *srcPath = std::getenv("SRC_PATH");
+
+	if (srcPath == NULL) {
+		cmdPath += ".";
+	} else {
+		cmdPath = srcPath;
+	}
+	cmdPath += "/cmd/" + zfsCmd + "/" + zfsCmd;
+
+	return cmdPath;
+}
+
+void execCmd(std::string zfsCmd, std::string args) {
+	int rc;
+	std::string cmdLine;
+
+	cmdLine = getCmdPath(zfsCmd) + " " + args;
+	rc = system(cmdLine.c_str());
+	if (rc != 0) {
+		throw std::runtime_error(
+		    std::string("Command failed: ") + cmdLine);
+	}
+}
+
+class CreatePool : public testing::Test {
+public:
+
+	spa_t *spa;
+	CreatePool() : zi(NULL), zv(NULL), spa(NULL) {}
+
+	virtual void SetUp() override {
+		uzfs_init();
+		libuzfs_ioctl_init();
+	}
+
+
+	void createPool() {
+		ASSERT_TRUE(uzfs_create_pool(pool_name, DISK, &spa ) == 0);
+	}
+
+	void importPool() {
+
+		char *spath[1];
+		nvlist_t *config = NULL;
+		nvlist_t *props = NULL;
+		importargs_t args = {0};
+
+		/*
+		 * we need to tell the import logic
+		 * where to find our truncated file
+		 * that it should use for import
+		 */
+
+		args.path = spath;
+		args.path[0] = "/tmp";
+		args.paths = 1;
+
+		args.scan = B_TRUE;
+		args.cachefile = NULL;
+
+		gzfs = libzfs_init();
+
+		ASSERT_TRUE(gzfs != NULL);
+
+		error = zpool_tryimport(gzfs, pool_name, &config, &args);
+		ASSERT_TRUE(error == 0);
+
+		error = spa_import(pool_name, config, props, ZFS_IMPORT_NORMAL);
+		ASSERT_TRUE(error == 0);
+
+		error = uzfs_open_pool(pool_name, &spa);
+		ASSERT_TRUE(error == 0);
+
+		libzfs_fini(gzfs);
+
+	}
+
+	void zinfoLookupByName() {
+
+		std::string zv_name(pool_name);
+		zv_name.append("/vol2");
+		zi = NULL;
+
+		zi = uzfs_zinfo_lookup(zv_name.c_str());
+		ASSERT_FALSE(zi == NULL);
+		zi->refcnt--;
+		ASSERT_EQ(zi->refcnt, 1);
+
+	}
+
+	void createDataset() {
+		 ASSERT_TRUE(uzfs_create_dataset(spa, "vol1", 64*1024*1024, 512, &zv) >= 0);
+		 /*  note our actual production code uses a different implementation */
+		 uzfs_zinfo_init(zv, zv->zv_name);
+	}
+
+	void getZVolInfo() {
+		/*  this bumps refcount  */
+		zi = uzfs_zinfo_lookup(zv->zv_name);
+		ASSERT_FALSE(zi == NULL);
+		ASSERT_EQ(strcmp(zi->name, zv->zv_name), 0);
+		zi->refcnt--;
+		ASSERT_EQ(zi->refcnt, 1);
+
+	}
+
+	void writeSome() {
+
+		 char *buf = (char* ) malloc(4096);
+		 memset(buf, 'j', 4096);
+		 ASSERT_EQ(uzfs_write_data(zi->zv, buf, 0, 4096, NULL, B_FALSE), 0);
+		 free(buf);
+	}
+
+	void readSome() {
+		char *buf = (char* ) malloc(4096);
+		memset(buf, 'a', 4096);
+		ASSERT_EQ(uzfs_read_data(zi->zv, buf, 0, 4096, NULL, 0), 0);
+		for (int i = 0; i < 4096; i++ )
+			ASSERT_EQ(buf[i], 'j');
+		free(buf);
+
+	}
+
+	void exportPool() {
+		uzfs_close_pool(spa);
+		ASSERT_EQ(spa_export(pool_name, NULL, B_TRUE, B_FALSE), 0);
+	}
+
+
+	void finish() {
```

Comment:
just a nit: this could be renamed to TearDown() - a hook called automatically at the end of test. That would nicely match with with SetUp().